http-gateways/SUBDOMAIN_GATEWAY.md

```diff
@@ -0,0 +1,164 @@
+# Subdomain Gateway Specification
+
+![](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square)
+
+**Authors**:
+
+- Marcin Rataj ([@lidel](https://github.com/lidel))
+
+----
+
+**Abstract**
+
+Subdomain Gateway is an extension of  [PATH_GATEWAY.md](./PATH_GATEWAY.md) that
+enables website hosting compatible with web browsers relative pathing and
+security model of the web. Below should be read as a delta on top of that spec.
+
+Summary:
+
+- data is requested by CID placed in `Host` header
+    - URL paths are not prefixed with `/ipfs/{cid}` or `/ipns/{foo}`
+- retrieve data from IPFS in a way that is compatible with URL-based addressing
+    - URL’s path `/` points at the content root identified by the CID
+- each CID is granted a unique [Origin sandbox](https://en.wikipedia.org/wiki/Same-origin_policy)
```

Comment:
```suggestion
- Requests carry the CID as a sub-domain in the `Host` header rather than as a URL path prefix
  - e.g. `{cid}.ipfs.example.org` instead of `example.org/ipfs/{cid}`
- The root CID is used to define the [Resource Origin](https://en.wikipedia.org/wiki/Same-origin_policy), aligning it with the web's security model.
  - Files in a DAG may request other files within the same DAG as part of the same Origin Sandbox.
- Data is retrieved from IPFS in a way that is compatible with URL-based addressing
    - URL’s path `/` points at the content root identified by the CID

```